feature/search/src/main/java/com/google/samples/apps/nowinandroid/feature/search/SearchViewModel.kt

```diff
@@ -16,15 +16,51 @@
 
 package com.google.samples.apps.nowinandroid.feature.search
 
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.samples.apps.nowinandroid.core.domain.GetSearchContentsUseCase
+import com.google.samples.apps.nowinandroid.feature.search.SearchResultUiState.LoadFailed
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
 import javax.inject.Inject
 
 @HiltViewModel
-class SearchViewModel @Inject constructor() : ViewModel() {
+class SearchViewModel @Inject constructor(
+    // TODO: Add GetSearchContentsCountUseCase to check if the fts tables are populated
+    getSearchContentsUseCase: GetSearchContentsUseCase,
+    private val savedStateHandle: SavedStateHandle,
+) : ViewModel() {
 
-    fun onSearchQueryChanged(searchQuery: String) {
-        // TODO: Pass it to UseCase
-        println("New search query is '$searchQuery'")
+    val searchQuery = savedStateHandle.getStateFlow("searchQuery", "")
+
+    val searchResultUiState: StateFlow<SearchResultUiState> =
+        searchQuery.flatMapLatest { query ->
+            if (query.length < 2) {
+                flowOf(SearchResultUiState.EmptyQuery)
+            } else {
+                getSearchContentsUseCase(query).map {
+                    SearchResultUiState.Success(
+                        topics = it.topics,
+                        newsResources = it.newsResources,
+                    )
+                }.catch {
+                    flowOf(LoadFailed)
```

Comment:
catch operator does not using return value of lambda, it uses `FlowCollector` as receiver
```suggestion
                    emit(LoadFailed)
```

Reply:
done.